Tests/IntegrationTests/Media+EdgeLocationHintTests.swift

```diff
@@ -116,11 +127,34 @@ class EdgeMediaLocationHintIntegrationTests: FunctionalTestBase {
         tracker.trackComplete()
 
         // verify
-        let networkRequests = getAllNetworkRequests()
+        mockNetworkService.assertAllNetworkRequestExpectations()
+        let networkRequests = mockNetworkService.getNetworkRequests()
         XCTAssertEqual(4, networkRequests.count)
         XCTAssertTrue(networkRequests[0].url.absoluteString.contains("https://edge.adobedc.net/ee/va/v1/sessionStart"))
         XCTAssertTrue(networkRequests[1].url.absoluteString.contains("https://edge.adobedc.net/ee/va/v1/play"))
         XCTAssertTrue(networkRequests[2].url.absoluteString.contains("https://edge.adobedc.net/ee/va/v1/pauseStart"))
         XCTAssertTrue(networkRequests[3].url.absoluteString.contains("https://edge.adobedc.net/ee/va/v1/sessionComplete"))
     }
+
+    /// Sets expectations for network requests based on specified media paths. Relative order
+    /// is **not** taken into account for assertions.
+    ///
+    /// This method counts the occurrences of each unique media path provided and
+    /// sets an expectation for a network request for each path with the respective count.
+    ///
+    /// - Parameter mediaPaths: A variadic parameter list of `MediaPath` values representing
+    ///   the total media paths for which network request expectations are to be set.
+    private func setExpectationForNetworkRequest(locationHint: String?, mediaEvents: XDMMediaEventType...) {
+        var pathCounts: [XDMMediaEventType: Int32] = [:]
+
+        for path in mediaEvents {
+            pathCounts[path, default: 0] += 1
+        }
+        for path in mediaEvents {
```

Comment:
Also update these iterator entries

'''
for event in mediaEvents {
    eventCount...
'''

Reply:
What if you also call in the first for loop. I think it will update the expectation map correctly right? Since setExpectationForNetworkRequest doesn't preserve order I am think its using a map underneath and it shouldn't be a problem if we update it multiple times it will eventually be set to a correct expected count.

```
for event in mediaEvents {
let url = "https://edge.adobedc.net/ee\(locationHint != nil ? "/\(locationHint!)" : "")/va/v1/" + event.rawValue

eventCount[event, default: 0] += 1
    
mockNetworkService.setExpectationForNetworkRequest(url: url, httpMethod: .post, expectedCount:  eventCount[event, default: 0] )
}
```